Make crate depend on core and alloc only

Dr-Emann opened the pull request:
Because of https://github.com/rust-lang/rust/issues/54010, `cargo test`
currently gives a spurious, harmless error in all but a very recent
nightly.

Dr-Emann committed:
Make crate depend on core and alloc only

Because of https://github.com/rust-lang/rust/issues/54010, `cargo test`
currently gives a spurious, harmless error in all but a very recent
nightly.

Lucretiel commented on src/lib.rs:
```diff
@@ -132,10 +134,13 @@
+extern crate alloc;
+
+use alloc::boxed::Box;
+use alloc::fmt;
```
This should be `core::fmt`, shouldn't it?

Dr-Emann replied:
Yes, it looks like core's fmt has all we need. Changed to `core::fmt`

Lucretiel reviewed (changes requested):
Thank you for the changes! I admit I wasn't very sure how `alloc` worked, which is why I didn't use it, at first.

When I run `cargo test` with these changes, I get the following error message when doctests run:

```
error: no global memory allocator found but one is required; link to std or add `#[global_allocator]` to a static item that implements the GlobalAlloc trait.

error: aborting due to previous error
```

Can you track down and fix the cause of this issue?

Lucretiel commented:
> Thank you for the changes! I admit I wasn't very sure how `alloc` worked, which is why I didn't use it, at first.
> 
> When I run `cargo test` with these changes, I get the following error message when doctests run:
> 
> ```
> error: no global memory allocator found but one is required; link to std or add `#[global_allocator]` to a static item that implements the GlobalAlloc trait.
> 
> error: aborting due to previous error
> ```
> 
> Can you track down and fix the cause of this issue?

Never mind, this appears to be a known bug in doctest.

Dr-Emann committed:
Use core::fmt rather than alloc's fmt

We don't need the parts of `fmt` that deal with Strings, so we can just
use the one from core

Lucretiel commented:
Thank you!
